src/renderer/components/NewInstanceModal.ts

```diff
@@ -0,0 +1,135 @@
+import Instance from "../Instance";
+import newInstanceModalTemplate from "../templates/NewInstanceModal.pug";
+import { InstanceData } from "../store/InstanceData";
+import { ApplicationStore } from "../store";
+import { Installer } from "@xmcl/installer";
+import InstanceStore from "../store/InstanceStore";
+
+export default class NewInstanceModal extends HTMLDivElement {
+	public constructor() {
+		super();
+	}
+	public connectedCallback(): void { }
+	public render(): void {
+		this.innerHTML = newInstanceModalTemplate();
+		$(this).modal("show");
+		this.attachEvents();
+	}
+	public attachEvents(): void {
+		// setup dropdowns
+		$(".ui.dropdown#dropdown-type").dropdown();
+		(document.getElementById("dropdown-type") as HTMLInputElement).addEventListener("change", () => {
+			this.updateIdDropdown($("#form-newInstance").form("get value", "instance-type"));
+		});
+		document.getElementById("submit-newInstanceForm")?.addEventListener("click", (e: MouseEvent) => {
+			e.preventDefault();
+			const $form = $("#form-newInstance");
+			$form.form("validate form");
+			if ($form.form("is valid")) {
+				// create new instance from input values
+				let tmpInstance: InstanceData = {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['tmpInstance' is never reassigned. Use 'const' instead.](https://app.codacy.com/manual/lukechu10/Minecraft-Box-Launcher/pullRequest?prid=5163001)